src/test/scala/JointSpec.scala

```diff
@@ -20,65 +19,175 @@ class JointSpec extends FunSuite {
+      phi=0, cohesion=0, shape=Nil)
+    assert(joint.dipDirection == math.Pi/2.0)
+  }
+
+  test("Dip direction for non-vertical joint should be 3.0*pi/4.0") {
+    val joint = Joint((-1.0/math.sqrt(3.0),-1.0/math.sqrt(3.0), 1.0/math.sqrt(3.0)), localOrigin=(0.0,0.0,-1.0), center=(0.0,0.0,0.0),
+      phi=0, cohesion=0, shape=Nil)
+    assert(joint.dipDirection == 3.0*math.Pi/4.0)
+  }
+
+  test("Dip direction for non-vertical joint should be 7.0*pi/4.0") {
+    val joint = Joint((1.0/math.sqrt(3.0), 1.0/math.sqrt(3.0), 1.0/math.sqrt(3.0)), localOrigin=(0.0,0.0,-1.0), center=(0.0,0.0,0.0),
+      phi=0, cohesion=0, shape=Nil)
+    assert(joint.dipDirection == 7.0*math.Pi/4.0)
+  }
+
+  test("Dip angle for horizontal joint should be 0.0") {
+    val joint = Joint((0.0, 0.0, 1.0), localOrigin=(0.0,0.0,-1.0), center=(0.0,0.0,0.0),
+      phi=0, cohesion=0, shape=Nil)
+    assert(joint.dipAngle == 0.0)
+  }
+
+  test("Dip angle for vertical joint should be pi/2.0") {
+    val joint = Joint((1.0, 0.0, 0.0), localOrigin=(0.0,0.0,-1.0), center=(0.0,0.0,0.0),
+      phi=0, cohesion=0, shape=Nil)
+    assert(joint.dipAngle == math.Pi/2.0)
+  }
+
+  test("Dip angle for joint with positive z-component in normal should be pi/6.0") {
+    val joint = Joint((1.0/math.sqrt(1.0 + math.sqrt(3.0)), 0.0, math.sqrt(3.0)/math.sqrt(1.0 + math.sqrt(3.0))), localOrigin=(0.0,0.0,-1.0), center=(0.0,0.0,0.0),
+      phi=0, cohesion=0, shape=Nil)
+    assert(math.abs(joint.dipAngle - math.Pi/6.0) < EPSILON)
+  }
+
+  test("Dip angle for joint with negative z-component in normal should be pi/6.0") {
+    val joint = Joint((0.0, 1.0/math.sqrt(1.0 + math.sqrt(3.0)), -math.sqrt(3.0)/math.sqrt(1.0 + math.sqrt(3.0))), localOrigin=(0.0,0.0,-1.0), center=(0.0,0.0,0.0),
+      phi=0, cohesion=0, shape=Nil)
+    assert(math.abs(joint.dipAngle - math.Pi/6.0) < EPSILON)
+  }
+
+  test("Dip direction of pi/2.0 centered at origin with positive z component in normal") {
+    val joint = Joint((0.0, 0.0, 1.0), localOrigin=(0.0,0.0,0.0), center=(0.0,0.0,0.0),
+                       phi=0, cohesion=0, shape=boundaries)
+    val newBoundaries: List[((Double,Double,Double), Double)] =
+        List(((1.0, 0.0, 0.0), 1.0),
+             ((-1.0, 0.0, 0.0), 0.0),
+             ((0.0, -1.0, 0.0), 1.0),
+             ((0.0, 1.0, 0.0), 0.0))
+    assert(totalDifference(joint.globalCoordinates, newBoundaries) <= EPSILON)
+  }
+
+  test("Dip direction of pi/2.0 centered at origin with negative z component in normal") {
+    val joint = Joint((0.0, 0.0, -1.0), localOrigin=(0.0,0.0,0.0), center=(0.0,0.0,0.0),
+      phi=0, cohesion=0, shape=boundaries)
     val newBoundaries: List[((Double,Double,Double), Double)] =
-        List(((0.0, -1.0, 0.0), 1.0),
-             ((0.0, 1.0, 0.0), 0.0),
-             ((1.0, 0.0, 0.0), 1.0),
-             ((-1.0, 0.0, 0.0), 0.0))
+      List(((1.0, 0.0, 0.0), 1.0),
```

Comment:
In case you're wondering, the way you had things indented here before is probably okay too.